### PR TITLE
Improve setCollateralToken, getCollateralToken and listCollateralTokens rpcs

### LINKED ISSUE #488: `DeFiCh/ain:2.x.x` Loan RPCs implementation
Similar to https://github.com/DeFiCh/jellyfish/issues/48, this tracks all the RPC methods and categories being implemented in `jellyfish-api-core`. Please tag and update this if you are working on a feature.

As a final docker build is not produced please use the latest `1.9.x` docker builds for your `testcontainers`, https://github.com/DeFiCh/jellyfish/pull/184 for more information on how to set custom image doing development.

### `rpc_loan.cpp`
- [x] createLoanScheme @jingyi2811 #520 
- [x] updateLoanScheme @jingyi2811 #530 
- [x] destroyLoanScheme @jingyi2811  #526 
- [x] setDefaultLoanScheme @jingyi2811  #554 
- [x] listLoanSchemes @jingyi2811  #524 
- [x] getLoanScheme @jingyi2811 #638 
- [x] setCollateralToken @jingyi2811  #537
- [x] getCollateralToken @jingyi2811  #546
- [x] listCollateralTokens @jingyi2811  #551
- [x] setLoanToken @jingyi2811  #568 
- [x] updateLoanToken @jingyi2811  #570 
- [x] getloantoken @surangap #758 
- [x] listLoanTokens @jingyi2811  #571 
- [x] takeLoan @canonbrother #661 
- [x] getLoanInfo @ivan-zynesis ~~#654~~ update: DeFiChain almost reworked the RPC, opened PR is invalid
- [x] loanPayBack @canonbrother #699 
- [x] getInterest @canonbrother #678
- [x] ~~getCollateralPrice~~

### `rpc_vault.cpp`
- [x] createVault @surangap  #553 
- [x] getVault  @surangap #569 
- [x] listVaults  @surangap #552 
- [x] updateVault @jingyi2811 #698 
- [x] depositToVault @canonbrother #661 
- [x] withdrawFromVault @ivan-zynesis #741  
- [x] closevault @jingyi2811 #730 
- [x] auctionBid @canonbrother #677 
- [x] listAuctions @jingyi2811 #695
- [x] listAuctionHistory @jingyi2811 #787 
- [ ] listVaultHistory @jingyi2811 #834
- [x] estimateloan @cwkang1998 #1347
- [x] estimatecollateral @cwkang1998 #1350 
- [x] estimatevault @cwkang1998 #1354 

### `rpc_oracles.cpp`
- [x] getfixedintervalprice @canonbrother  #760 
- [x] listfixedintervalprices @canonbrother #760 

### `rpc_poolpair.cpp`
- [x] compositeswap @ivan-zynesis #776

### `mn_rpc.cpp`
- [x] setGovHeight @ivan-zynesis #789
- [x] listGovs@ivan-zynesis #789
- [x] ~~getGov~~ @ivan-zynesis  #794 defid reverted

---
Known flaky test for investigation:
~~https://github.com/DeFiCh/jellyfish/pull/752/checks?check_run_id=3930860973#step:5:3610~~

### PR DESCRIPTION
**Pull request by jingyi2811**
#### What kind of PR is this?:

/kind fix

#### What this PR does / why we need it:
Fix the texts that was broken by the latest ain code 

#### Which issue(s) does this PR fixes?:

Fixes #488

### CODE REVIEW

**Review by surangap — COMMENTED** [review]
@jingyi2811 what about the tests in `packages/jellyfish-api-core/__tests__/category/loan/getCollateralToken.test.ts`  and related refactor?

**Comment by jingyi2811** [comment]
> @jingyi2811 what about the tests in `packages/jellyfish-api-core/__tests__/category/loan/getCollateralToken.test.ts` and related refactor?

I have parked them in other PRs

**Comment by jingyi2811** [comment]
For getCollateralToken, looks like it still cannot accept id like 1,2 ... 10 as a parameter. The C++ programmer is still working on this.

File: packages/jellyfish-api-core/__tests__/category/loan/loan_container.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { MasterNodeRegTestContainer, StartOptions, GenesisKeys, MasterNodeKey }
 
 export class LoanMasterNodeRegTestContainer extends MasterNodeRegTestContainer {
   constructor (masternodeKey: MasterNodeKey = GenesisKeys[0]) {
-    super(masternodeKey, 'defi/defichain:HEAD-fffc8dd')
+    super(masternodeKey, 'defi/defichain:HEAD-5c9779c')
```
Comment (canonbrother) [inline packages/jellyfish-api-core/__tests__/category/loan/loan_container.ts:5]: can try the latest one?

```suggestion
    super(masternodeKey, 'defi/defichain:HEAD-39c3669')
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 2]
```diff
@@ -111,22 +111,23 @@ export class Loan {
   /**
    * List collateral tokens.
    *
+   * @param {ListCollateralTokens} [collateralToken = {}]
+   * @param {number} [collateralToken.height = CurrentBlockheight] Valid at specified height
+   * @param {boolean} [collateralToken.all] True = All transactions, false =  Activated transactions
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:116]: one more filtered by token

File: docs/node/RPC Category/16-loan.md [hunk 3]
```diff
@@ -152,7 +152,12 @@ List collateral tokens.
 
 ```ts title="client.loan.listCollateralTokens()"
 interface loan {
-  listCollateralTokens (): Promise<CollateralTokensData>
+  listCollateralTokens (collateralToken: ListCollateralTokens = {}): Promise<CollateralTokensData>
```
Comment (jingyi2811) [inline docs/node/RPC Category/16-loan.md:155]: I am unsure when they will fix this.  They promise me that listCollateralTokens won't accept token as parameter anymore.

File: packages/jellyfish-api-core/src/category/loan.ts (hunk 2 above)
Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/loan.ts:116]: I am unsure when they will fix this. They promise me that listCollateralTokens won't accept token as parameter anymore.

Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/loan.ts:116]: Only getCollateralToken will accept token as parameter

Comment (canonbrother) [reply packages/jellyfish-api-core/src/category/loan.ts:116]: weird.. because there has token filter.. will that be removed??

Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/loan.ts:116]: I shall text them again

Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/loan.ts:116]: Okay, He just told me that he would fix this by today. He was busy with something else.

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 4]
```diff
@@ -111,22 +111,23 @@ export class Loan {
...
+    return await this.client.call('listcollateraltokens', [collateralToken], 'bignumber')
   }
 
   /**
    * Get collateral token.
    *
-   * @param {GetCollateralToken} [collateralToken = {}]
-   * @param {string} [collateralToken.token] Symbol of collateral token
-   * @param {number} [collateralToken.height = CurrentBlockheight] Valid at specified height
+   * @param {string} token
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/loan.ts:126]:
```suggestion
   * @param {string} token symbol or id
```

**Comment by canonbrother** [comment]
specific the param desc then overall is good for me

File: packages/jellyfish-api-core/__tests__/category/loan/listCollateralTokens.test.ts [hunk 5]
```diff
@@ -56,11 +50,89 @@ describe('Loan listCollateralTokens', () => {
...
+          factor: new BigNumber(1),
+          priceFeedId: 'TSLA/USD',
+          token: 'TSLA'
+        }
+      })
+    }
+  })
+})
+
+describe('Loan listCollateralTokens with height', () => {
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/listCollateralTokens.test.ts:89]: This tests, add another collateral token with height. then show what it returns with `height`.
Also can add a test to show that if `all` and `height` combination used, only `all` is considered.

**Review by surangap — COMMENTED** [review]
request to add few more tests. https://github.com/DeFiCh/jellyfish/pull/689/files#r721018352
the rest looks good.

File: packages/jellyfish-api-core/__tests__/category/loan/listCollateralTokens.test.ts (hunk 5 above)
Comment (jingyi2811) [reply packages/jellyfish-api-core/__tests__/category/loan/listCollateralTokens.test.ts]: Fixed

**Review by canonbrother — APPROVED** [review]

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**